pkgs/development/libraries/qt-6/default.nix

```diff
@@ -65,6 +65,8 @@ let
             url = "https://github.com/Homebrew/formula-patches/raw/c363f0edf9e90598d54bc3f4f1bacf95abbda282/qt/qt_internal_check_if_path_has_symlinks.patch";
             sha256 = "sha256-Gv2L8ymZSbJxcmUijKlT2NnkIB3bVH9D7YSsDX2noTU=";
           })
+          # this patch requires postprocessing, see postPatch in qtbase.nix
+          ./patches/qtbase-qmake-qt-prepare-tool.diff
```

Comment:
```suggestion
          (substituteAll {
            src = ./patches/qtbase-qmake-qt-prepare-tool.diff
            NIXPKGS_QTBASE_FIND_QT_TOOL = ./patches/qtbase-qmake-qt-prepare-tool-find-qt-tool.sh
          })
```

Reply:
this does not work

find-qt-tool.sh requires patching of shebang and `chmod +x`
[substituteAll](https://github.com/milahu/nixpkgs/blob/8f0c1b40c5f3350ae62e29d2f06a9c5d4994f89e/pkgs/stdenv/generic/setup.sh#L941) replaces only `@camelCase@` variables (or `@snake_case@`)

```sh
# Substitute all environment variables that start with a lowercase character and
# are valid Bash names.
substituteAll() {
```

<details>

```sh
#!/usr/bin/env bash

tempdir=$(mktemp -d)
(set -x; cd $tempdir)

cat >default.nix <<'EOF'
{ pkgs ? import <nixpkgs> {} }:
with pkgs;
stdenv.mkDerivation {
name = "test";
src = substituteAll {
  src = ./main.sh;
  HELLO_SH = ./hello.sh;
  hello_sh = ./hello.sh;
};
buildCommand = ''
install -Dm 555 $src $out/bin/main.sh
patchShebangs $out/bin
'';
}
EOF

cat >main.sh <<'EOF'
#! /bin/sh
${HELLO_SH}
@@HELLO_SH@@
@HELLO_SH@
@hello_sh@
EOF

cat >hello.sh <<'EOF'
#! /bin/sh
echo hello
EOF

nix-build .

head -v -n999 result/bin/main.sh

cat result/bin/main.sh |
grep -E -o '/nix/store/[0-9a-z]{32}-hello.sh' |
xargs head -v -n999

echo mode:
cat result/bin/main.sh |
grep -E -o '/nix/store/[0-9a-z]{32}-hello.sh' |
xargs stat -c"%a %n"

rm -rf $tempdir
```

```
+ cd /tmp/tmp.ttyznasDKN
/nix/store/rwb0r2kpi7qraqnwrhidcva8mcd9yjrg-test
==> result/bin/main.sh <==
#!/nix/store/y0g1mvsr6vygr61f9znljik9kl0x0inc-bash-5.1-p16/bin/sh
${HELLO_SH}
@@HELLO_SH@@
@HELLO_SH@
/nix/store/ci5vm0h3zi2v79qqh6zqwhzx8gqlb86z-hello.sh
==> /nix/store/ci5vm0h3zi2v79qqh6zqwhzx8gqlb86z-hello.sh <==
#! /bin/sh
echo hello
mode:
444 /nix/store/ci5vm0h3zi2v79qqh6zqwhzx8gqlb86z-hello.sh
```

</details>

but this works, and its obviously better than my `postPatch`

```nix
(substituteAll {
  src = ./patches/qtbase-qmake-qt-prepare-tool.diff;
  findQtTool = writeShellScript "find-qt-tool.sh" (
    builtins.readFile ./patches/qtbase-qmake-qt-prepare-tool-find-qt-tool.sh);
})
```